crates/sp-consensus-subspace/src/inherents.rs

```diff
@@ -16,90 +16,119 @@
 
 //! Inherents for Subspace consensus
 
-use sp_inherents::{Error, InherentData, InherentIdentifier};
-
+use codec::{Decode, Encode};
+use sp_consensus_slots::Slot;
+use sp_inherents::{Error, InherentData, InherentIdentifier, IsFatalError};
+use sp_runtime::RuntimeDebug;
 use sp_std::result::Result;
+use sp_std::vec::Vec;
+use subspace_core_primitives::RootBlock;
 
 /// The Subspace inherent identifier.
 pub const INHERENT_IDENTIFIER: InherentIdentifier = *b"subspace";
 
+/// Errors that can occur while checking root blocks.
+#[derive(Encode, RuntimeDebug)]
+#[cfg_attr(feature = "std", derive(Decode))]
+pub enum InherentError {
+    /// List of root blocks is not correct.
+    IncorrectRootBlocksList,
+}
+
+impl IsFatalError for InherentError {
+    fn is_fatal_error(&self) -> bool {
+        true
+    }
+}
+
 /// The type of the Subspace inherent data.
-pub type SubspaceInherentData = sp_consensus_slots::Slot;
+#[derive(Encode, Decode, RuntimeDebug)]
+pub struct InherentType {
```

Comment:
Why not `SubspaceInherentData`? Don't think `InherentType` is more reabable.

Reply:
I went back and forth on this, it makes more sense this way and here is why: `SubspaceInherentData` usage here means that what was previously called `SubspaceInherentData` needs to be renamed into something different now. You used `SubspaceInherentDataTrait`, which would make sense it it was implemented for `SubspaceInherentData`, but it is implemented for `InherentData` instead. So it makes way more sense to have `impl SubspaceInherentData for InherentData` and what is inside of it to be called `InherentType`, comparing to `impl SubspaceInherentDataTrait for InherentData` and `SubspaceInherentData` being a different thing inside of it.

So I very much agree with the naming convention of Substrate here, they use the same naming in their crates. Renaming this breaks consistency and while looks nice, non-intuitiveness of naming makes it much harder to follow IMO.

Reply:
I'm not totally convinced. Using `InherentType` just sounds like a foreign word in this context.

We should try to follow the convention(the namings, project structure, etc) as much as we can, if not perfectly the same, choose a way that looks the same if possible, for that will definitely save you some time and mental cost eventually because we have to work on their codebase(if you can't change it, you get used to it), even it might be just one or two hours at the end of the day.

Reply:
The mental model is the following: `InherentData` is already provided by substrate. Internally it contains a `BTreeMap` where keys are `[u8; 8]` and values are `Vec<u8>` of something that is encodable. In this file the key is `INHERENT_IDENTIFIER`, which is specific to Subspace and `InherentType` is the type of the inherent that was encoded into `Vec<u8>` before being placed into mentioned `BTreeMap`. `SubspaceInherentData` as renamed is just a Subspace-specific adapter into `InherentData` that gets `INHERENT_IDENTIFIER` key and tries to decode `Vec<u8>` returned as `InherentType`.

I hope with this description it is a bit clearer why above text would look weird if we replace `InherentType` with `SubspaceInherentData` and `SubspaceInherentData` with `SubspaceInherentDataTrait`, it will no longer look as consistent in naming.